front-end/src/hooks/authContext.js

```diff
@@ -1,72 +1,87 @@
+import Civic from "../services/Civic";
+import { getUserStorage, setUserStorage, clearUserStorage } from "./storage";
+import { clearProposalsStorage } from "./storage";
 
 let civic = new Civic();
 const authContext = createContext();
 
 function useProvideAuth() {
-    const [isLoggedInValue, setIsLoggedIn] = useState(false);
-    const [isGov, setIsGov] = useState(false);
+  const [isLoggedInValue, setIsLoggedIn] = useState(false);
+  const [isGov, setIsGov] = useState(false);
 
-    async function isLoggedIn() {
-        if (isLoggedInValue === true) return true;
+  async function isLoggedIn() {
+    if (isLoggedInValue === true) return true;
 
-        const user = getUserStorage();
-        if (user) {
-            await civic.accountLoginWithKey(user.accountName, user.commonName, user.privKey)
-            setIsGov(user.isGov);
-            setIsLoggedIn(true);
-            return true;
-        } else {
-            return isLoggedInValue;
-        }
+    const user = getUserStorage();
+    if (user) {
+      await civic.accountLoginWithKey(
+        user.accountName,
+        user.commonName,
+        user.privKey
+      );
+      setIsGov(user.isGov);
+      setIsLoggedIn(true);
+      return true;
+    } else {
+      return isLoggedInValue;
     }
+  }
 
-    async function login(accountName, password) {
-        const login = await civic.accountLogin(accountName, password);
-        setUserStorage(accountName, civic.account.commonName, civic.account.privateKey, login.isGov);
-        setIsGov(login.isGov);
-        setIsLoggedIn(true);
-    }
+  async function login(accountName, password) {
+    const login = await civic.accountLogin(accountName, password);
+    setUserStorage(
+      accountName,
+      civic.account.commonName,
+      civic.account.privateKey,
+      login.isGov
+    );
+    setIsGov(login.isGov);
+    setIsLoggedIn(true);
+  }
 
-    async function createAccount(accountName, password, commonName) {
-        const create = await civic.accountCreate(accountName, password, commonName);
-        setUserStorage(accountName, commonName, civic.account.privateKey, create.isGov);
-        setIsGov(create.isGov);
-        setIsLoggedIn(true);
-    }
+  async function createAccount(accountName, password, commonName) {
+    const create = await civic.accountCreate(accountName, password, commonName);
+    setUserStorage(
+      accountName,
+      commonName,
+      civic.account.privateKey,
+      create.isGov
+    );
+    setIsGov(create.isGov);
+    setIsLoggedIn(true);
+  }
 
-    async function logout() {
-        clearUserStorage();
-        setIsLoggedIn(false);
-    }
+  async function logout() {
+    clearUserStorage();
+    clearProposalsStorage();
+    setIsLoggedIn(false);
+  }
 
-    return {
-        civic,
-        isGov,
-        isLoggedIn,
-        login,
-        createAccount,
-        logout
-    };
+  return {
+    civic,
+    isGov,
+    isLoggedIn,
+    isLoggedInValue,

```

Comment:
why is this needed now? isLoggedIn() returns this

Reply:
isLoggedIn() only returns true or false
civic instance is necessary for the civic calls outside the context
I am also using isLoggedInValue state to update proposals on the voteContext